workflow-core/src/test/java/com/squareup/workflow/WorkerTest.kt

```diff
@@ -1,41 +1,111 @@
+/*
+ * Copyright 2020 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 package com.squareup.workflow
 
 import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.buffer
+import kotlinx.coroutines.flow.conflate
 import kotlinx.coroutines.flow.emptyFlow
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.toList
+import kotlinx.coroutines.runBlocking
+import kotlin.reflect.typeOf
 import kotlin.test.Test
+import kotlin.test.assertEquals
 import kotlin.test.assertFalse
+import kotlin.test.assertNotEquals
+import kotlin.test.assertNotSame
 import kotlin.test.assertTrue
 
+@OptIn(ExperimentalStdlibApi::class)
 class WorkerTest {
 
-  @Test fun `default Worker#doesSameWorkAs implementation compares by concrete type`() {
-    class Worker1 : Worker<Nothing> {
-      override fun run(): Flow<Nothing> = emptyFlow()
-    }
-
-    class Worker2 : Worker<Nothing> {
-      override fun run(): Flow<Nothing> = emptyFlow()
-    }
-
-    assertTrue(Worker1().doesSameWorkAs(Worker1()))
-    assertFalse(Worker1().doesSameWorkAs(Worker2()))
-  }
-
   @Test fun `createSideEffect workers are equivalent`() {
     val worker1 = Worker.createSideEffect {}
```

Comment:
Will deprecate this method in a follow-up PR, since side effects are a primary concept now.

Reply:
#114